federation/plugin/pkg/admission/schedulingpolicy/admission.go

```diff
@@ -37,21 +39,34 @@ const (
 	pluginName               = "SchedulingPolicy"
 	configKey                = "schedulingPolicy"
 	policyConfigMapNamespace = "kube-federation-scheduling-policy"
-	minRetryBackoff          = time.Millisecond
-	maxRetryBackoff          = time.Second * 60
-	defaultRetryBackoff      = time.Millisecond * 100
+
+	// Lower and upper bounds on policy engine query retry delays. The actual
+	// backoff implementation is handled by k8s.io/apiserver/pkg/util/webhook.
+	// The defaultRetryBackoff is used if the admission controller
+	// configuration does not specify one.
+	minRetryBackoff     = time.Millisecond
+	maxRetryBackoff     = time.Second * 60
+	defaultRetryBackoff = time.Millisecond * 100
 )
 
 type admissionConfig struct {
-	Kubeconfig   string        `json:"kubeconfig"`
-	RetryBackoff time.Duration `json:"retryBackoff"`
+	Kubeconfig                   string        `json:"kubeconfig"`
+	RetryBackoff                 time.Duration `json:"retryBackoff"`
+	FailOpenIfPolicyDoesNotExist bool          `json:"failOpenIfPolicyDoesNotExist"`
 }
 
 type admissionController struct {
 	*admission.Handler
 	policyEngineClient       *rest.RESTClient            // client to communicate with policy engine
 	policyEngineRetryBackoff time.Duration               // backoff for policy engine queries
-	client                   internalclientset.Interface // client to communicate with federation-apiserve
+	client                   internalclientset.Interface // client to communicate with federation-apiserver
+
+	// controls whether policy engine is queried when policy has not been
```

Comment:
The admin would set this flag to true if they were loading the policies into the engine out-of-band and wanted to exercise the policies for test purposes.

This flag would not be set in production.

In the initial version of the PR this flag was not necessary as the admission controller ALWAYS queried the policy engine and the fail-open vs fail-closed decision was determined by the existence of a ConfigMap in the built-in namespace.